### PR TITLE
Add test to make sure we don't suggest duplicate suggestion

### PR DESCRIPTION
**Pull request by fabiencelier**
## Description

closes https://github.com/dust-tt/tasks/issues/7181

 - slightly change how we assert the tool call result to be more detailed
 - Add 2 new tests to make sure we don't produce new suggestion when we already have a similar suggestion which is pending or rejected.

## Tests

```
 RUN  v4.0.18 /Users/fabiencelier/spaces/dust/.hives/ra--duplicate/front

 ✓ tests/reinforced-agent-evals/reinforced-agent-eval.test.ts (8 tests) 52450ms
   ✓ Reinforced Agent Evaluation Tests (8)
     ✓ analyze-conversation (3)
       ✓ bad-tone-user-requests-warmth  19447ms
       ✓ missing-websearch-skill  20284ms
       ✓ missing-jira-tool  27075ms
     ✓ aggregate-suggestions (5)
       ✓ three-jira-tool-suggestions  15236ms
       ✓ three-websearch-skill-suggestions  16079ms
       ✓ three-tone-prompt-suggestions  25313ms
       ✓ no-duplicate-of-pending-prompt-suggestion  14121ms
       ✓ no-duplicate-of-rejected-prompt-suggestion  13339ms

 Test Files  1 passed (1)
      Tests  8 passed (8)
   Start at  11:29:38
   Duration  57.00s (transform 1.23s, setup 373ms, import 3.77s, tests 52.45s, environment 332ms)
```

## Risk

low, still feature flagged

## Deploy Plan

deploy front https://github.com/dust-tt/tasks/issues/7181

### CODE REVIEW

File: front/tests/reinforced-agent-evals/lib/assertions.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    }
+    case "promptSuggestion": {
+      const call = toolCalls.find((tc) => tc.name === "suggest_prompt_edits");
+      if (!call) {
+        return {
+          success: false,
+          error: "Expected suggest_prompt_edits to be called, but it was not",
+        };
+      }
+      const suggestions = getSuggestions(call.arguments);
```
Comment (davidebbo) [inline front/tests/reinforced-agent-evals/lib/assertions.ts]: Should it optionally validate the targetBlockId? That should be quite deterministic.

File: front/tests/reinforced-agent-evals/test-suites/analyze-conversation.ts
Comment (davidebbo) [inline front/tests/reinforced-agent-evals/test-suites/analyze-conversation.ts:36]: We could have some test cases with multiple blocks and block IDs to match reality, and make sure the suggestions are attached to the correct block.

**Review by davidebbo — APPROVED** [review]
LGTM. Couple comments are for future improvements.

File: front/tests/reinforced-agent-evals/lib/assertions.ts (hunk 1 above)
Comment (fabiencelier) [reply front/tests/reinforced-agent-evals/lib/assertions.ts]: Good idea, I'll add that for future test
Not adding such test right now though
